### PR TITLE
LBAC-41: LBAC On/Off

### PR DESCRIPTION
**Pull request by VANKINEENITAWRUN**
## Issue I worked on

see https://issues.openmrs.org/browse/LBAC-41

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by suthagar23** [comment]
@VANKINEENITAWRUN Is it working? Can you attach a screenshot of the responses?

And follow JIRA guidelines also

**Comment by VANKINEENITAWRUN** [comment]
@suthagar23  added tests to check its working

**Comment by dkayiwa** [comment]
@suthagar23 what do you think of this?

**Comment by VANKINEENITAWRUN** [comment]
@suthagar23  its not the error with the code. please check it.
https://travis-ci.community/t/error-installing-oraclejdk8-expected-feature-release-number-in-range-of-9-to-14-but-got-8/3766

File: omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LBACOnOffController.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/lbac/on-off")
+public class LBACOnOffController extends BaseRestController {
+
+	@RequestMapping(method = RequestMethod.POST)
```
Comment (suthagar23) [inline omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LBACOnOffController.java]: Can't we use postMapping? 
https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/bind/annotation/PostMapping.html

File: omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LBACOnOffController.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	@ResponseBody
+	@ResponseStatus(value = HttpStatus.OK)
+	public void changeEntityGlobalProperty(@RequestBody Map<String, String> body) {
+		List<String> entities = getLocationEntites();
+		for (String entity : entities) {
+			String entityGlobalPropertyValue = body.get(entity);
+			if (entityGlobalPropertyValue != null) {
```
Comment (suthagar23) [inline omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LBACOnOffController.java]: Can't we use stringUtils?

File: omod/src/test/java/org/openmrs/module/locationbasedaccess/web/controller/LBACOnOffController_Test.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	private String getURI() {
+		return "lbac/on-off";
+	}
+
+	@Test
+	public void changeEntityGlobalProperty_shouldChangeGlobalPropertyValue() throws Exception {
+		Assert.assertEquals(administrationService.getGlobalProperty(gp1), "true");
+		Assert.assertEquals(administrationService.getGlobalProperty(gp2), "false");
+		Assert.assertEquals(administrationService.getGlobalProperty(gp3), "false");
+		SimpleObject obj = new SimpleObject();
```
Comment (suthagar23) [inline omod/src/test/java/org/openmrs/module/locationbasedaccess/web/controller/LBACOnOffController_Test.java]: Add new lines after each set of tests to keep them as separated.

File: omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LBACOnOffController.java (hunk 1 above)
Comment (VANKINEENITAWRUN) [reply omod/src/main/java/org/openmrs/module/locationbasedaccess/web/controller/LBACOnOffController.java]: @suthagar23  The postMapping Annotation is from 4.3 the one we are using currently is 4.1.4.

**Pull request merged**